pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -47,6 +47,11 @@ func (p *PoolManager) AllocateVirtualMachineMac(virtualMachine *kubevirt.Virtual
 		return nil
 	}
 
+	// We can't allow for duplicate interfaces names, as interface.Name is macMap's key.
+	if err := checkVmForInterfaceDuplication(virtualMachine); isNotDryRun && err != nil {
```

Comment:
We don't want to even run `checkVmForInterfaceDuplication` if we are on dry mode right ?

Reply:
yeah, with inlining I meant that we should do something like `if !isDryRun && ...`

Reply:
@RamLavi can you fix it ?

Reply:
I don't think it's a working syntax. if we want to not run `checkVmForInterfaceDuplication` if not needed  then either we do but don't return the err from `checkVmForInterfaceDuplication`:
```
if isNotDryRun && checkVmForInterfaceDuplication(virtualMachine) != nil {
    ...
}
```
or do  it in a 2 lines like initially done
```
if isNotDryRun {
    if err := checkVmForInterfaceDuplication(virtualMachine); err != nil {
        return err
    }
}
```
do you have another option?

Reply:
Yes, the latter (without in-lining) seems preferable. Sorry for dragging you into this.

Reply:
no worries, better it be as clean as possible :) 

even better, moved the flag inside the function on @qinqon 's suggestion. what do you think?

Reply:
I think boolean arguments suck. The caller knows what to do: call the function, or avoid doing so, based on the value of the boolean.

No point passing it around. 

My 2 cents :)

Reply:
Let's not inline the boolean check then and do as @RamLavi proposed

Reply:
[Done](https://github.com/k8snetworkplumbingwg/kubemacpool/compare/1ea79aa610967f9f4b42500bcfc7377e7ed1dafc..53f7cda95cc5b39942d4b6662a256682ec13ff49)